src/mo_frontend/typing.ml

```diff
@@ -1362,85 +1363,96 @@ let resolve_hole env at hole_sort typ =
     | T.{ lab; typ = Typ c; _ } -> None
     | T.{ lab; typ = Mut t; _ } -> None
     | T.{ lab = lab1; typ = typ1; _ } ->
-       if is_matching_lab lab1 &&
-          is_matching_typ typ1
+       if is_matching_typ typ1
        then Some (lab1, typ1)
        else None
   in
-  let find_candidate (module_name, fs) =
-    List.find_map has_matching_field_typ fs |>
-      Option.map (fun (lab, typ)->
+  let find_candidate_fields (module_name, fs) =
+    List.filter_map has_matching_field_typ fs |>
+      List.map (fun (lab, typ)->
           let path =
-            { it = DotE( { it = VarE {it = module_name; at = no_region; note = Const};
-                           at = Source.no_region;
-                           note = empty_typ_note
-                         },
-                         { it = lab; at = no_region; note = () },
-                         ref None);
+            { it = DotE(
+                { it = VarE {it = module_name; at = no_region; note = Const};
+                  at = Source.no_region;
+                  note = empty_typ_note
+                },
+                { it = lab; at = no_region; note = () },
+                ref None);
               at = Source.no_region;
               note = empty_typ_note; }
           in
-          ({ path; desc = module_name^"."^ lab; typ } : hole_candidate))
+          ({ path; desc = module_name^"."^ lab; typ; id=lab } : hole_candidate))
   in
-  let find_candidate_val = function
+  let find_candidate_id = function
     (id, (t, _, _, _)) ->
-    if is_matching_lab id &&
-       is_matching_typ t
+    if is_matching_typ t
     then
-      let path = { it =
-                   VarE {it = id; at = no_region; note = Const};
-                   at = Source.no_region;
-                   note = empty_typ_note }
+      let path =
+        { it = VarE {it = id; at = no_region; note = Const};
+          at = Source.no_region;
+          note = empty_typ_note }
       in
-      Some { path; desc = id; typ = t }
+      Some { path; desc = id; typ = t; id }
     else None
   in
-  let eligible_env_vals =
+  let (eligible_ids, explicit_ids) =
     let vals =
       match hole_sort with
       | Named id ->
-        (* narrow env to search *)
-        (match T.Env.find_opt id env.vals with
-         | Some info -> T.Env.singleton id info
-         | None -> T.Env.empty)
+         env.vals
       | Anon _ ->
-         env.vals (* search entire env *)
+         env.vals
     in
```

Comment:
It should be just `let vals = env.vals` now
I cannot make a GH suggestion unfortunately